### PR TITLE
Solution for error #274

### LINKED ISSUE #274: [13.0] sale_commission_pricelist - sale_commission_salesman error
**Reproduced in [runbot](http://3454218-13-0-d7a0c7.runbot2-2.odoo-community.org/web?#action=288&model=sale.order&view_type=list&cids=&menu_id=170)**

**Steps followed:**
1. Install sale_commission_pricelist and sale_commission_salesman
2. In the administrator contact (Mitchel Admin) check "Agent" in the "Sales & Purchase" tab.
3. Check "Convert salesman into agent" in the "Agent Information" tab.
4. In a pricelist add two new product lines and add to each one the commission (different commissions)
5. Create sales order (Commercial: Mitchel Admin; Pricelist: the pricelist mentioned in the previous point). Add two order lines: the first with product "FURN_6666", the second with product "E-COM11".

**When you add the second order line the following error appears** 
Odoo Server Error
Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/api.py", line 745, in get
    value = self._data[field][record._ids[0]]
KeyError: <NewId ref='virtual_224'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/fields.py", line 1004, in __get__
    value = env.cache.get(record, self)
  File "/.repo_requirements/odoo/odoo/api.py", line 751, in get
    raise CacheMiss(record, field)
odoo.exceptions.CacheMiss: ("sale.order.line(<NewId ref='virtual_224'>,).agent_ids", None)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/models.py", line 5096, in ensure_one
    _id, = self._ids
ValueError: too many values to unpack (expected 1)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/http.py", line 624, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/.repo_requirements/odoo/odoo/http.py", line 310, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/.repo_requirements/odoo/odoo/tools/pycompat.py", line 14, in reraise
    raise value
  File "/.repo_requirements/odoo/odoo/http.py", line 669, in dispatch
    result = self._call_function(**self.params)
  File "/.repo_requirements/odoo/odoo/http.py", line 350, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/service/model.py", line 94, in wrapper
    return f(dbname, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/http.py", line 339, in checked_call
    result = self.endpoint(*a, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 915, in __call__
    return self.method(*args, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 515, in response_wrap
    response = f(*args, **kw)
  File "/.repo_requirements/odoo/addons/web/controllers/main.py", line 1327, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/.repo_requirements/odoo/addons/web/controllers/main.py", line 1319, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 387, in call_kw
    result = _call_kw_multi(method, model, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 374, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/models.py", line 6251, in onchange
    for name in nametree
  File "/.repo_requirements/odoo/odoo/models.py", line 6252, in <listcomp>
    if name not in done and snapshot0.has_changed(name)
  File "/.repo_requirements/odoo/odoo/models.py", line 6097, in has_changed
    return self[name] != record[name]
  File "/.repo_requirements/odoo/odoo/models.py", line 5726, in __getitem__
    return self._fields[key].__get__(self, type(self))
  File "/.repo_requirements/odoo/odoo/fields.py", line 999, in __get__
    self.compute_value(recs)
  File "/.repo_requirements/odoo/odoo/fields.py", line 1113, in compute_value
    records._compute_field_value(self)
  File "/.repo_requirements/odoo/odoo/models.py", line 3998, in _compute_field_value
    getattr(self, field.compute)()
  File "/home/odoo/build/OCA/commission/sale_commission/models/sale_order.py", line 13, in _compute_commission_total
    record.commission_total = sum(record.mapped("order_line.agent_ids.amount"))
  File "/.repo_requirements/odoo/odoo/models.py", line 5302, in mapped
    recs = recs._mapped_func(operator.itemgetter(name))
  File "/.repo_requirements/odoo/odoo/models.py", line 5261, in _mapped_func
    vals = [func(rec) for rec in self]
  File "/.repo_requirements/odoo/odoo/models.py", line 5261, in <listcomp>
    vals = [func(rec) for rec in self]
  File "/.repo_requirements/odoo/odoo/models.py", line 5726, in __getitem__
    return self._fields[key].__get__(self, type(self))
  File "/.repo_requirements/odoo/odoo/fields.py", line 2968, in __get__
    return super().__get__(records, owner)
  File "/.repo_requirements/odoo/odoo/fields.py", line 2333, in __get__
    return super().__get__(records, owner)
  File "/.repo_requirements/odoo/odoo/fields.py", line 1028, in __get__
    self.compute_value(recs)
  File "/.repo_requirements/odoo/odoo/fields.py", line 1113, in compute_value
    records._compute_field_value(self)
  File "/.repo_requirements/odoo/odoo/models.py", line 3998, in _compute_field_value
    getattr(self, field.compute)()
  File "/home/odoo/build/OCA/commission/sale_commission_salesman/models/sale_order.py", line 18, in _compute_agent_ids
    record.agent_ids = [(0, 0, self._prepare_agent_vals(partner))]
  File "/home/odoo/build/OCA/commission/sale_commission_pricelist/models/sale_order.py", line 34, in _prepare_agent_vals
    self.ensure_one()
  File "/.repo_requirements/odoo/odoo/models.py", line 5099, in ensure_one
    raise ValueError("Expected singleton: %s" % self)
ValueError: Expected singleton: sale.order.line(NewId ref='virtual_224', NewId ref='virtual_239')

### PR DESCRIPTION
**Pull request by sinerkiaid**
Solves [13.0] sale_commission_pricelist - sale_commission_salesman error

Fixes #274

### CODE REVIEW

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Fast-tracking it:

/ocabot merge patch

File: sale_commission_salesman/models/sale_order.py [hunk 1]
```diff
@@ -15,4 +15,4 @@ def _compute_agent_ids(self):
         for record in self.filtered(lambda x: x.order_id.partner_id):
             partner = self.order_id.user_id.partner_id
```
Comment (bodedra) [inline sale_commission_salesman/models/sale_order.py]:
```suggestion
            partner = record.order_id.user_id.partner_id
```

Comment (pedrobaeza) [reply sale_commission_salesman/models/sale_order.py]: True as well... aborting merge

**Review by bodedra — APPROVED** [review]
LG :+1:

**Pull request merged**